api/lib/infrastructure/repositories/correction-repository.js

```diff
@@ -1,44 +1,69 @@
+const _ = require('lodash');
+
 const Correction = require('../../domain/models/Correction');
 const Hint = require('../../domain/models/Hint');
 const challengeDatasource = require('../datasources/airtable/challenge-datasource');
 const skillDatasource = require('../datasources/airtable/skill-datasource');
-
-function _convertSkillDataObjectsToHints(skillDataObjects) {
-  return skillDataObjects.map((skillDataObject) => {
-    return new Hint({
-      skillName: skillDataObject.name,
-      value: skillDataObject.hint
-    });
-  });
-}
+const tutorialDatasource = require('../datasources/airtable/tutorial-datasource');
 
 function _getSkillDataObjects(challengeDataObject) {
   const skillDataObjectPromises = challengeDataObject.skillIds.map(skillDatasource.get);
   return Promise.all(skillDataObjectPromises);
 }
 
+function _getTutorialDataObjects(tutorialIds) {
+  const tutorialDataObjectPromise = tutorialIds.map(tutorialDatasource.get);
+  return Promise.all(tutorialDataObjectPromise);
+}
+
 function _selectSkillDataObjectsWithValidatedHint(skillDataObjects) {
   return skillDataObjects.filter((skillDataObject) => skillDataObject.hintStatus === 'Validé');
 }
 
+function _convertSkillDataObjectsToHints(skillDataObjects) {
+  return skillDataObjects.map((skillDataObject) => {
+    return new Hint({
+      skillName: skillDataObject.name,
+      value: skillDataObject.hint
+    });
+  });
+}
+
 module.exports = {
 
   getByChallengeId(challengeId) {
     let challengeDataObject;
+    let hintsForChallenge;
+    let skillsForChallenge;
+    let tutorialsIds;
 
     return challengeDatasource.get(challengeId)
       .then((retrievedChallengeDataObject) => {
         challengeDataObject = retrievedChallengeDataObject;
         return retrievedChallengeDataObject;
       })
       .then(_getSkillDataObjects)
-      .then(_selectSkillDataObjectsWithValidatedHint)
+      .then((skills) => {
+        skillsForChallenge = skills;
+        return _selectSkillDataObjectsWithValidatedHint(skillsForChallenge);
```

Comment:
```js
return skills.filter(_selectSkillDataObjectsWithValidatedHint)
```
proposition faible

Reply:
Why not, on laisse comme avant quand même pour garder la function à part et bien séparer les actions sur les objets